### PR TITLE
docs(components): Adding Grouped Components to side nav bar [JOB-110760]

### PR DESCRIPTION
**Pull request by Beccaneer**
## Notes

- Styling will be completed in JOB-110489
- The /components path doesn't load for some reason, I'm not sure if its a problem with my branch
- I have no idea why the dark mode css file is being added?

## Changed

Allowing components to be grouped under sections in the side nav bar
- Added `sections` parameter to objects in `componentList`
- Made `path` in `AtlantisRoute` optional to accommodate for section headings
- Added section heading component in left side nav
- Added `AnimatedPresenceDisclosure` component to replace `Disclosure` for top level nav menu items with children
     - Top level items can have children that are links (like Changelog) as well as children that are section headings (like Components). The children that are section headings can have children that are links. 
     - In the Components example, the route structure is: 
           
                     { 
                       handle: Top Level Item, 
                        ... ,
                         children: [ 
                                { handle: sectionHeading1, NO PATH, ..., children: [ { ComponentA }, { ComponentB }, { ComponentC }]}
                                { handle: sectionHeading2, NO PATH, ..., children: [ { ComponentD }, { ComponentE }, { ComponentF }]}
                            ],
                          }
           
- Expanded logic in `NavMenu.tsx` and `Layout.tsx` to accommodate for children of children in `AtlantisRoute`

## Testing

Run the docs site from the packages/site directory with `npm run dev`.
Expand 'Components' and you should see all components under their respective section headings.

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

Screenshot 2024-12-06 at 9 08 37 AM

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by taylorvnoj** [comment]
@Beccaneer There is something going on - I do see the individual components in the Components disclosure, I can click on the individual components however none of the main side nav links work for me.  Home, etc don't load. The console errors are so continuous I can't stop to read them 😅 Can you click on other main sidenav links?
We'll definitely need to see what's going on in this branch because `master` seems ok

Screenshot 2024-12-06 at 2 53 38 PM

**Comment by Beccaneer** [comment]
@taylorvnoj  It's pointing to a line that was added by Scott earlier today, I think I might need to get him to look at it :\

**Comment by taylorvnoj** [comment]
@Beccaneer ah ok - yeah I think that's best. Otherwise the groupings and individual component pages are looking good

File: packages/site/src/componentList.ts [hunk 1]
```diff
@@ -3,166 +3,192 @@ export const componentList = [
     title: "ActionItem",
     to: "/components/ActionItem",
     imageURL: "/ActionItem.png",
-    additionalMatches: [],
+    sections: ["Layouts and Structure"],
```
Comment (taylorvnoj) [inline packages/site/src/componentList.ts:6]: I really like the way you handled this!

**Review by taylorvnoj — APPROVED** [review]
Looks great and works really well on the site - main collections pages are loading as expected as are individual components. We know we'll do design updates as a follow up, as well as look into `InputCurrency` and `ActivityIndicator` 👀 
Great job @Beccaneer :shipit:

**Pull request merged**